experimental/design-notes.txt

```diff
@@ -0,0 +1,112 @@
+as increment() and set(). Meters are associated with a single set of tags.
+For the OT spec Meters are factories that create instruments and it is the instruments that have the data collecting APIs such as Add() or Record(). Instruments
+aren't associated with a specific set of tags (labels) unless they are first bound.
+As best I can tell the best conceptual mapping is that MM's Meter == OT's bound instrument. It doesn't worry me too much that OT would use a different
+name for the same concept, what worries me more is if OT is reusing the same names but altering the concept they apply to.
+
+2. MM has no equivalent abstraction for OT's Meter or Instrument. It goes straight from MeterProvider (MeterRegistry in MM-terminology) to a bound-instrument 
+(Meter in MM-terminology). There is an even simpler case for people using a default global registry where they can use static functions defined in the Metrics class.
+If MM has been succesful with these simpler abstractions it raises the question what value is OT getting from its more complex 4 layers of
+abstraction (MeterProvider->Meter->Instrument->BoundInstrument)?
+
+3. MM landed on a different set of instrument types than OT did. It would be good to understand why we believe OT's instrument set is the right set to pick?
+MM has Timer, Counter, Gauge, DistributionSummary, LongTaskTimer, FunctionCounter, FunctionTimer, and TimeGauge. I think the conceptual match up is roughly:
+Timer                   NA
+Counter                 Counter
+DistributionSummary     ValueRecorder
+Gauge                   ValueObserver
+LongTaskTimer           NA
+FunctionCounter         SumObserver
+FunctionTimer           NA
+TimeGauge               NA
+NA                      UpDownCounter
+NA                      UpDownSumObserver
+
+
+4. MM has a different model for observation/asynchronicity than OT. MM has the Meter (in MM terminology) provide a callback function that when invoked returns
+the value of the measurement for that measurement interval. OT also uses a callback but the callback is expect to invoke a Observe() API to record potentially
+many data points and has void return type. I think the additional complexity in the OT model is a side-effect of the design decision to have unbound instruments
+that might need to set the value of potentially many LabelSets whereas in MM all Meters (in MM terminology) are bound to a single set of labels and will
+have exactly 1 data point per time period. Does the value of unbound instruments justify this additional complexity?
+
+5. MM has no batch mode for logging metrics, how valuable of a feature is this given its additional complexity to implement? Is it purely a performance optimization
+or is there a scenario where the atomicity of the measurements matter?
+
+6. MM has a lot of API surface dedicated to configuring Meters (in MM terminology) and tags. I am guessing this is the kind of functionality that would be sitting in OT's
+Views API?
```

Comment:
This will be a HUGE topic for SDK side.  And probably will be very vendor opinionated in those discussions!